### DISCUSSION TITLE
Setting inter-paragraph spacing?

### DISCUSSION THREAD

**Discussion by user202729** [typst/typst#318] [Q&A]
As in the title. How can I do that? Intuitively it should be in the documentation of `par` or `parbreak` or `page` but I can't find anything.

**Comment by mateusfccp** [comment]
I tried doing so by overwriting `par` with `#show par: it => ...`, but the compiler in the web kept crashing.

In my case, I want something even more specific. I want to be able to put a vertical space (`#v(...)`) after the last paragraph of a section. I don't even know if we have a conception of section, but it would be basically the last paragraph before a new nth-level heading, being `n > 1`.

**Reply by ararunaufc** [reply]
Would setting a space _before_ the heading be a feasible workaround? Maybe setting `weak: true` so it doesn't take effect when at page start.

**Reply by mateusfccp** [reply]
I don't think so, because it would sum to the spacing that I need after the heading whenever I nest headings.

**Reply by laurmaedje** [reply]
Weak spacing shouldn't sum.

**Reply by mateusfccp** [reply]
This seems to work:

```
#show heading: it => [
  #set text(size: 12pt)

  #v(3em, weak: true)
  #counter(heading).display() #if(it.level == 1) [#(upper(it.body))] else [#it.body]
  #v(3em, weak: true)
]
```

I suggest being more clear what flow is... I've read about the `weak` parameter in the documentation:

> If true, the spacing collapses at the start or end of a flow. 

However, I have no idea what it is a flow...

Thanks for the hint!

**Comment by ararunaufc** [comment]
If I'm not mistaken, `par`s are laid out using `block`, so you could do something like:

```typst
#show par: set block(below: 1em)
```

**Reply by mateusfccp** [reply]
I am going to give it a try

**Comment by AuRobinson** [comment]
I couldn't get it to work,
```typst
#show par: set block(below: 1em)
```
Quick solution for me has been `#v()`, although it's a bit tedious. 

This issue reminds me of the gutter parameter in [`#grid()`](https://typst.app/docs/reference/layout/grid/). Don't know if it's related though.

**Reply by xkevio** [reply]
No, your show-rule works as-is and is the idiomatic/correct way to solve this. Your document might have some other issue or show-rule preventing this one from applying properly then.

**Reply by AuRobinson** [reply]
I'll have another look then.

**Reply by AuRobinson** [reply]
@xkevio -- Much appreciated.

https://typst.app/docs/reference/styling/#show-rules
